packages/flutter/lib/src/material/radio.dart

```diff
@@ -447,7 +447,10 @@ class _RadioState<T> extends State<Radio<T>> with TickerProviderStateMixin, Togg
               value: widget.value,
               groupValue: widget.groupValue,
               onChanged: widget.onChanged,
-              mouseCursor: widget.mouseCursor,
+              mouseCursor: WidgetStateProperty.resolveWith((Set<MaterialState> states) {
+                return MaterialStateProperty.resolveAs<MouseCursor?>(widget.mouseCursor, states)
+                  ?? CupertinoRadio.defaultMouseCursor(widget.onChanged);
+              }),
```

Comment:
If CupertinoRadio.defaultMouseCursor returns a WidgetStateProperty then could you do this instead?

```dart
              mouseCursor: widget.mouseCursor == null
                ? CupertinoRadio.defaultMouseCursor(widget.onChanged)
                : WidgetStateProperty.all<MouseCursor>(widget.mouseCursor!),
```

Reply:
This works, I forgot we could just use `!` to keep the analyzer quiet. Thanks for the insight!